src/amazons3/amazons3_errors.bal

```diff
@@ -17,70 +17,149 @@
 # Holds the details of an AmazonS3 error
 #
 # + message - Specific error message for the error
-# + errorCode - Error code for the error
 # + cause - Cause of the error; If this error occurred due to another error (Probably from another module)
 public type ErrorDetail record {
     string message;
-    string errorCode;
     error cause?;
 };
 
-// Ballerina AmazonS3 Client Error Types
-
-public const CLIENT_CONFIG_INITIALIZATION_FAILED = "{wso2/amazons3}ClientConfigInitializationFailed";
-public type ClientConfigInitializationFailed error<CLIENT_CONFIG_INITIALIZATION_FAILED, ErrorDetail>;
+// Client error reasons
```

Comment:
Not sure if reasons is the right word here. Is it types or reasons based on types?

Reply:
These are reasons for respective client error types.